### PR TITLE
Fixed duplicates in functions name

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/29991

## Description
Fixed the functions duplication names.
Manage to run the entire flow and upload successfully to Demisto.

## Screenshots
image

## Must have
- [x] Tests

### CODE REVIEW

**Comment by Shellyber** [comment]
Working on test

File: demisto_sdk/commands/openapi_codegen/openapi_codegen.py [hunk 1]
```diff
@@ -1009,7 +1008,7 @@ def clean_function_name(name: str, snakeify: bool = True) -> str:
 
         if snakeify:
             name = camel_to_snake(name)
-            name = name.replace('-', '_').replace('__', '_').strip('_')
+            name = name.replace('-', '_').replace('__', '_').replace('.', '').strip('_')
```
Comment (liorblob) [inline demisto_sdk/commands/openapi_codegen/openapi_codegen.py:1011]: I think`.` should be in `ILLEGAL_CODE_CHARS`

File: demisto_sdk/commands/openapi_codegen/openapi_codegen.py [hunk 2]
```diff
@@ -1032,3 +1031,24 @@ def format_params(params: list, base: str, base_string: str) -> str:
...
+        if duplicate_names:
+            for func in functions:
+                name = func.get('name')
+                path = func.get('path')
+                method = func.get('method')
+                # getting the last curly brackets is exists and keeping its value
+                function_path = re.sub(r'{([\w]*)\}$', r'\g<1>', path)
+                # Remove the rest curly brackets from the path.
+                path_name = '_'.join([re.sub(r'{[^)]*\}', '', x) for x in function_path.split('/')])
+                if name in duplicate_names:
```
Comment (liorblob) [inline demisto_sdk/commands/openapi_codegen/openapi_codegen.py:1052]: this `if` should be right after you get the name

File: demisto_sdk/commands/openapi_codegen/openapi_codegen.py [hunk 3]
```diff
@@ -1032,3 +1031,24 @@ def format_params(params: list, base: str, base_string: str) -> str:
 
         params_string = base.replace(base_string, ', '.join(modified_params))
         return params_string
+
+    def change_name_duplications(self, functions):
```
Comment (liorblob) [inline demisto_sdk/commands/openapi_codegen/openapi_codegen.py:1035]:
```suggestion
    def handle_duplicates(self, functions):

```

File: demisto_sdk/tests/test_files/swagger_python.txt [hunk 4]
```diff
@@ -159,6 +159,28 @@ class Client(BaseClient):
...
+
+        headers = self._headers
+        headers['Content-Type'] = 'multipart/form-data'
+
+        response = self._http_request('post', f'pet/{petId}/uploadImage', json_data=data, headers=headers)
+
+        return response
+
+
+    def post_pet_upload_image_uploadimage_request(self, petId, additionalMetadata, file):
```
Comment (liorblob) [inline demisto_sdk/tests/test_files/swagger_python.txt:173]: the name looks weird. is it supposed to be like that?

**Review by liorblob — CHANGES_REQUESTED** [review]

File: demisto_sdk/tests/test_files/swagger_python.txt (hunk 4 above)
Comment (Shellyber) [reply demisto_sdk/tests/test_files/swagger_python.txt:173]: yes, its the only way to handle this kind of duplications.

File: demisto_sdk/commands/openapi_codegen/openapi_codegen.py (hunk 1 above)
Comment (Shellyber) [reply demisto_sdk/commands/openapi_codegen/openapi_codegen.py]: It's wasn't there but I added it.

**Comment by Shellyber** [comment]
Some commands examples:
```
'newtest-get-config-authority-router-routing-ospf-area-interface': 
'newtest-get-config-authority-router-routing-ospf-area-interface-by-interface': 
'newtest-get-config-authority-router-routing-ospf-redistribute': 
'newtest-get-config-authority-router-routing-ospf-redistribute-by-redistribute': 
'newtest-get-config-authority-router-routing-routing-protocol-address-family': 
'newtest-get-config-authority-router-routing-routing-protocol-address-family-by-addressfamily'\
```

**Comment by Shellyber** [comment]
@liorblob I'm done. Please review and approve

File: demisto_sdk/commands/openapi_codegen/openapi_codegen.py [hunk 5]
```diff
@@ -1032,3 +1031,24 @@ def format_params(params: list, base: str, base_string: str) -> str:
 
         params_string = base.replace(base_string, ', '.join(modified_params))
         return params_string
+
+    def handle_duplicates(self, functions):
```
Comment (Itay4) [inline demisto_sdk/commands/openapi_codegen/openapi_codegen.py:1035]: please add type hints

File: demisto_sdk/commands/openapi_codegen/openapi_codegen.py (hunk 2 above)
Comment (liorblob) [reply demisto_sdk/commands/openapi_codegen/openapi_codegen.py:1052]: @Shellyber please address

**Review by liorblob — APPROVED** [review]

**Pull request merged**